### DISCUSSION TITLE
SQL install and update from Templates

### DISCUSSION THREAD

**Discussion by leoalv** [joomla/joomla-cms#42496] [General]
Is it possible to add or make changes to the database from a template as is done from other types of extensions using the following?
```
<install>
    <sql>
        <file driver="mysql" charset="utf8">sql/install/mysql/install.sql</file>
    </sql>
</install>
<update>
    <schemas>
        <schemapath type="mysql">sql/updates/mysql</schemapath>
        <schemapath type="sqlsrv">sql/updates/sqlsrv</schemapath>
    </schemas>
</update>
```

I know, this is not common and seems unscientific, but the fact is that I have fields of type REPEATABLE (predecessor of subform), I have made the queries to migrate and to make things easier I want it to be able to be migrated when updating the template.

Regards

**Comment by alikon** [comment]
preflight() and postflight() in  script.php doesn't work ?

**Reply by leoalv** [reply]
Hi @alikon.
I think it should work, I also think that using the manifest would be more manageable.
On the other hand, my curiosity is to know why what works for components, modules and plugins does not work in templates if they are also extensions and the documentation does not detail them.

Thanks for your answer.

**Reply by alikon** [reply]
> why what works for components, modules and plugins does not work in templates

if true, than it's a bug
i've never developed/tested a template scenario
